### PR TITLE
Efrat/fix purge

### PR DESCRIPTION
**Pull request by efrat-lb**
on the pulling from osmosis branch 
fixed issue regarding purge of unstale hashes found by @yuval-lb

### CODE REVIEW

File: cpp/Osmosis/main.cpp [hunk 1]
```diff
@@ -171,7 +171,7 @@ void purge( const boost::program_options::variables_map & options )
 
 	// adding task per dir
 	for(auto& entry : boost::make_iterator_range(boost::filesystem::directory_iterator( rootPath ), {})) {
-		if ( entry.path().string().length() == 31) {
+		if ( entry.path().stem().string().length() == 2) {
```
Comment (yuval-lb) [inline cpp/Osmosis/main.cpp]: @efrat-lb what is "stem" - last directory in path?

File: cpp/Osmosis/ObjectStore/Purge.cpp [hunk 2]
```diff
@@ -21,8 +21,10 @@ void Purge::purge( boost::filesystem::path & dirToPurge )
 	size_t after = _staleHashes.size();
 	TRACE_INFO( "Purge found " << after << " objects to purge (" <<
 			( before - after ) << " remain)" );
-	for ( auto & hash : _staleHashes )
-		boost::filesystem::remove( _store.filenameForExisting( hash ) );
+	for ( auto & hash : _staleHashes ){
```
Comment (yuval-lb) [inline cpp/Osmosis/ObjectStore/Purge.cpp]: whats the motive here?

File: cpp/Osmosis/main.cpp (hunk 1 above)
Comment (efrat-lb) [reply cpp/Osmosis/main.cpp]: yes. (can use filename() as well)

File: cpp/Osmosis/ObjectStore/Purge.cpp (hunk 2 above)
Comment (efrat-lb) [reply cpp/Osmosis/ObjectStore/Purge.cpp]: export the exists from filenameForExisting() -
 cpp/Osmosis/ObjectStore/Store.cpp :line 39 `ASSERT( exists( hash ) );`

**Pull request closed without merging**